File: src/main/java/com/matag/game/turn/action/attach/AttachAction.java
package com.matag.game.turn.action.attach;

import static java.lang.Integer.parseInt;

import org.springframework.stereotype.Component;

import com.matag.game.cardinstance.CardInstance;
import com.matag.cards.ability.AbilityAction;
import com.matag.cards.ability.Ability;
import com.matag.game.status.GameStatus;

import lombok.AllArgsConstructor;

@Component
@AllArgsConstructor
public class AttachAction implements AbilityAction {

  private final AttachService attachService;

  @Override
  public void perform(CardInstance cardInstance, GameStatus gameStatus, Ability ability) {
    var target = cardInstance.getModifiers().getTargets().get(0).toString();
    var attachedToId = parseInt(target);
    attachService.attach(gameStatus, cardInstance, attachedToId);
  }
}

File: src/main/kotlin/com/matag/game/turn/action/attach/AttachAction.kt
package com.matag.game.turn.action.attach

import com.matag.cards.ability.Ability
import com.matag.cards.ability.AbilityAction
import com.matag.game.cardinstance.CardInstance
import com.matag.game.status.GameStatus
import org.springframework.stereotype.Component

@Component
class AttachAction(
    private val attachService: AttachService
) : AbilityAction {
    override fun perform(cardInstance: CardInstance, gameStatus: GameStatus, ability: Ability?) {
        val target = cardInstance.modifiers.targets[0].toString()
        val attachedToId: Int = target.toInt()
        attachService.attach(gameStatus, cardInstance, attachedToId)
    }
}
